src/Parser/CommandInfo.php

```diff
@@ -639,6 +639,10 @@ public function addArgumentDescription($name, $description, $suggestedValues = [
         $this->addOptionOrArgumentDescription($this->arguments(), $name, $description, $suggestedValues);
     }
 
+    public function addOptionDescriptionDefaultValue($name, $description, $suggestedValues = [], $defaultValue = null) {
```

Comment:
Reviewing this in context with the rest of the file, it became clear that this "new" method is just `addOptionDescription` with the addition of a default value parameter. (n.b. "findMatchingOption" returns its input `$name` parameter if it cannot find the option in question.)

My suggestion:
1. Add a defaultValue parameter to `addOptionDescription`, and make this method deprecated.
2. Just name the`addOptionDescriptionDefaultValue` method `addOption`. This is the way folks should add options to a CommandInfo when working with code that participates in Annotation parsing or Attribute handling.
3. `addOption` just calls through to `addOptionDescription` (or the other way around, if you prefer); these methods are now synonyms of each other, we're just renaming the existing method to something that's more readable / straightforward for the Attribute use case.
4. In cases where Attributes are calling `addOption`, we expect that `$this->arguments()->removeMatching()` should not return a default value. (If it did, it would indicate that someone used an attribute to add an option, say, `--ssh-options`, when there is already a command argument named `ssh-options`.) If it does, then we should just use the passed-in value, and ignore $defaultFromParameter. In other words, the final line should be `$this->addOptionOrArgumentDescription($this->options(), $variableName, $description, $suggestedValues, $defaultValue ?? $defaultFromParameter);`.

Does that all sound reasonable to you?

Reply:
I updated this PR and Drush PR https://github.com/drush-ops/drush/pull/5443